src/clients/github.js

```diff
@@ -14,26 +14,55 @@ const log = getActiveLogger();
  * @returns {Error}
  */
 function filterErrorResponse(err) {
+  const filteredError = new Error();
   if (err && err.response) {
     // GitHub API response.text returns a stringified JSON that needs to be parsed
-    const filteredError = {
-      status: err.response.status,
-      text: JSON.parse(err.response.text),
-      req: {}
-    };
+    filteredError.status = err.response.status;
+    filteredError.text = err.response.text;
+    filteredError.req = {};
     if (err.response.req) {
-      filteredError.req = {
-        method: err.response.req.method,
-        url: err.response.req.url
-      };
+      filteredError.req.method = err.response.req.method;
+      filteredError.req.url = err.response.req.url;
     }
-    return new Error(JSON.stringify(filteredError));
+    return filteredError;
   }
-
   return err;
 }
 
 /**
+ * Delete GitHub OAuth token.
+ *
+ * @param githubUrl
+ * @param username
+ * @param password
+ * @returns {Promise}
+ */
+export async function deleteGitHubToken(githubUrl, username, password) {
+  log.verbose('deleting github token');
+  log.debug('deleteGitHubToken', { username });
+
+  if (githubUrl) {
+    return new Promise((resolve, reject) => {
+      superagent
+        .delete(githubUrl)
+        .set({ Authorization: authorizationUtil.basicAuthorization(username, password) })
+        .end((err) => {
+          if (err) {
+            if (err.response.status !== 404) {
+              console.log('Already deleted');
+              reject(filterErrorResponse(err));
```

Comment:
I think if the token is already deleted, you should call `resolve()` instead of `reject()`, since what we wanted to accomplish was already accomplished. Or am I misunderstanding what this is used for?

Reply:
However, if it's not already deleted (if there was some other error), then we need to reject. So there should be an else block with `reject()`.

Reply:
I just didn't want it to do anything basically.  I should at least change it to log.info... 
Currently, we delete it in two places, if the app executes successfully.  I guess I should be able to remove one of them.  and then remove this.  I'll do that.

Reply:
I changed my mind.  I'm not going to delete either deleteGitHubTokenCall because one of them is inside the initProjects, the other is in the finally block around the call to initProjects.  The second is in case any errors are thrown, the githubToken will still be deleted. The first time it is called is at the end of initProjects. This is when it should normally be deleted. I don't want to remove it from this method in case tests or future methods use initProjects.

Can I reject without throwing an error? It should be fine if it receives a 404 because the token has already been deleted

Reply:
If the finally block catches both instances, then I would remove the other instance. But I would still change it to resolve() if you find it's already deleted.